### PR TITLE
class library: fix bug in Server tcp code.

### PR DESCRIPTION
**Pull request by miguel-negrao**

### CODE REVIEW

**Comment by miguel-negrao** [comment]
Should fix #3246.

**Comment by LFSaw** [comment]
looking good (no error anymore) but still getting this:

```
-> others
booting server 'others' on address: localhost:57113
NetAddr-Connect failed with exception: connect: Connection refused
Found 102 LADSPA plugins
NetAddr-Connect failed with exception: connect: Connection refused
Number of Devices: 8
   0 : "Built-in Microph"
   1 : "Built-in Output"
   2 : "Soundflower (2ch)"
   3 : "Soundflower (64ch)"
   4 : "Instant On Sound Effects"
   5 : "REAPER device"
   6 : "BFH5"
   7 : "BF-UC"

"Built-in Microph" Input Device
   Streams: 1
      0  channels 2

"Built-in Output" Output Device
   Streams: 1
      0  channels 2
```

**Comment by LFSaw** [comment]
and server is still stuck in "booting" mode.

File: SCClassLibrary/Common/Control/Server.sc [hunk 1]
```diff
@@ -861,7 +861,7 @@ Server {
 			this.disconnectSharedMemory;
 			pid = unixCmd(program ++ options.asOptionsString(addr.port), { statusWatcher.quit(watchShutDown:false) });
 			("booting server '%' on address: %:%").format(this.name, addr.hostname, addr.port.asString).postln;
-			if(options.protocol == \tcp, { addr.tryConnectTCP(onComplete) }, onComplete);
+			if(options.protocol == \tcp, { addr.tryConnectTCP(10) }, onComplete);
```
Comment (LFSaw) [inline SCClassLibrary/Common/Control/Server.sc]: I guess you mean 

```
addr.tryConnectTCP(maxAttempts: 10)
```

but this does not help either...

**Comment by miguel-negrao** [comment]
Sorry, I think I has versions of the files opened from different commits. I think now this should be the correct fix. Can you try ?

**Comment by miguel-negrao** [comment]
Btw, it's normal to get a couple of connection error messages until we manage to connect to the server:

```
booting server 'others' on address: localhost:57110
NetAddr-Connect failed with exception: connect: Connection refused
Found 2 LADSPA plugins
NetAddr-Connect failed with exception: connect: Connection refused
JackDriver: client name is 'SuperCollider'
SC_AudioDriver: sample rate = 44100.000000, driver's block size = 512
JackDriver: connected  system:capture_1 to SuperCollider:in_1
JackDriver: connected  system:capture_2 to SuperCollider:in_2
JackDriver: connected  SuperCollider:out_1 to system:playback_1
JackDriver: connected  SuperCollider:out_2 to system:playback_2
SuperCollider 3 server ready.
```

**Comment by LFSaw** [comment]
this works.

**Pull request merged**